### PR TITLE
Feature/config updates

### PR DESCRIPTION
**Pull request by jbygdell**
### Describe the pull request:

- [X] Bug fix
- [X] Functional change
- [ ] New feature
- [X] Code cleanup
- [ ] Build system change
- [X] Documentation change
- [ ] Language translation

### Pull request long description:

This i a major update for the helm chart and it is done in preparation for upstream changes to among other things new docker images.

### Changes made:

1. Separate image entries for each of the services.
2. All options in the `defaults.ini` file are now available as ENV.
3. Secrets and config maps can no longer be excluded during deployment.
4. S3 related configs have been renamed to better reflect their position.

### Release note:

Version bumped to `v0.2.0`

### CODE REVIEW

File: ega-charts/localega/templates/mapper-deploy.yaml [hunk 1]
```diff
@@ -25,27 +25,44 @@ spec:
...
       containers:
       - name: mapper
-        image: "{{ .Values.base.image }}:{{ .Values.base.imageTag }}"
-        imagePullPolicy: {{ .Values.base.imagePullPolicy | quote }}
+        image: "{{ .Values.mapper.repository }}:{{ .Values.mapper.imageTag }}"
+        imagePullPolicy: {{ .Values.mapper.imagePullPolicy | quote }}
         command: ["gosu", "lega", "ega-id-mapper"]
         env:
+        - name: DEFAULT_LOG
+          value: "debug"
```
Comment (blankdots) [inline ega-charts/localega/templates/mapper-deploy.yaml]: I see in other places this used: ` value: {{ .Values.config.log | quote }}`

File: ega-charts/localega/templates/mapper-deploy.yaml [hunk 2]
```diff
@@ -25,27 +25,44 @@ spec:
...
+          value: "/"
         - name: POSTGRES_HOST
           value: {{ if .Values.config.postgres_host }}{{ .Values.config.postgres_host | quote }}{{ else }}{{ template "localega.fullname" . }}-db{{ end }}
         - name: POSTGRES_PASSWORD
           valueFrom:
             secretKeyRef:
               name: {{ template "localega.name" . }}-db-secret
               key: postgres_password
+        - name: POSTGRES_SSLMODE
+          value: "disable"
```
Comment (blankdots) [inline ega-charts/localega/templates/mapper-deploy.yaml]: maybe a  miss on this: `{{ .Values.config.broker_enable_ssl | quote }}`

File: ega-charts/localega/templates/verify-deploy.yaml [hunk 3]
```diff
@@ -26,56 +26,84 @@ spec:
...
       containers:
       - name: verify
-        image: "{{ .Values.base.image }}:{{ .Values.base.imageTag }}"
-        imagePullPolicy: {{ .Values.base.imagePullPolicy | quote }}
+        image: "{{ .Values.verify.repository }}:{{ .Values.verify.imageTag }}"
+        imagePullPolicy: {{ .Values.verify.imagePullPolicy | quote }}
         command: ["gosu", "lega", "ega-verify"]
         env:
+        - name: DEFAULT_LOG
+          value: "debug"
```
Comment (blankdots) [inline ega-charts/localega/templates/verify-deploy.yaml]: `value: {{ .Values.config.log | quote }}`

File: ega-charts/localega/templates/verify-deploy.yaml [hunk 4]
```diff
@@ -26,56 +26,84 @@ spec:
...
+          value: "/"
         - name: POSTGRES_HOST
           value: {{ if .Values.config.postgres_host }}{{ .Values.config.postgres_host | quote }}{{ else }}{{ template "localega.fullname" . }}-db{{ end }}
         - name: POSTGRES_PASSWORD
           valueFrom:
             secretKeyRef:
               name: {{ template "localega.name" . }}-db-secret
               key: postgres_password
+        - name: POSTGRES_SSLMODE
+          value: "disable"
```
Comment (blankdots) [inline ega-charts/localega/templates/verify-deploy.yaml]: `{{ .Values.config.broker_enable_ssl | quote }}`

File: ega-charts/localega/values.yaml [hunk 5]
```diff
@@ -1,30 +1,39 @@
...
 # Declare variables to be passed into your templates.
-base:
-  image: nbisweden/ega-base
-  imageTag: latest
-  imagePullPolicy: Always
 
 config:
-  deploy: true
-  broker_host: ""
+  log: "default"
```
Comment (blankdots) [inline ega-charts/localega/values.yaml]: I propose to use `debug` as default.

File: README.md [hunk 6]
```diff
@@ -67,30 +67,42 @@ Parameter | Description | Default
...
-`config.cega_endpoint` | URL to CEGA user service | `""`
-`config.cega_endpoint_json` | Prefix in the JSON response that contains the user info| `response.result`
+`config.broker_username` | username for accessing local MQ service | `guest`
+`config.cega_host` | hostname of CEGA users service | `""`
+`config.cega_endpoint` | URL to CEGA user service | `"/lega/v1/legas/users/%s?idType=username"`
+`config.cega_endpoint_json` | Prefix in the JSON response that contains the user info | `response.result`
 `config.cega_mq_host` | CEGA MQ hostname | `""`
 `config.cega_vhost` | CEGA MQ vhost path | `""`
-`config.cega_port` | CEGA MQ acces port | `5271`
+`config.cega_port` | CEGA MQ acces port | `5272`
```
Comment (blankdots) [inline README.md]: I think this is the default port when connecting to CRG ? but it should be `5672` i assume most people will use the default RabbitMQ - but not a hard requirement

In the `values.yml` i see it is `5672`

**Review by blankdots — CHANGES_REQUESTED** [review]
Review: first part see comments.

I have yet to test the charts, and I will do so in 2nd part.

File: ega-charts/localega/templates/verify-deploy.yaml (hunk 4 above)
Comment (jbygdell) [reply ega-charts/localega/templates/verify-deploy.yaml]: No those are different, this is if we need SSL for the connectin to the postgres sever.

File: ega-charts/localega/templates/mapper-deploy.yaml (hunk 2 above)
Comment (jbygdell) [reply ega-charts/localega/templates/mapper-deploy.yaml]: This is a part that is not fully implemented yet as out postgres don't accept ssl connections as it is.

Comment (blankdots) [reply ega-charts/localega/templates/mapper-deploy.yaml]: My bad ... I think I meant to write this: `{{ .Values.config.postgres_sslmode | quote }}` in `ingest-deploy` I think it should be everywhere the same.

File: ega-charts/localega/templates/verify-deploy.yaml (hunk 4 above)
Comment (blankdots) [reply ega-charts/localega/templates/verify-deploy.yaml]: `{{ .Values.config.postgres_sslmode | quote }}` - mistake meant this :disappointed:

**Review by jonandernovella — CHANGES_REQUESTED** [review]
When working with string data, specially with secrets, it's  highly recommended **quoting the strings**. Don't do it with integers though. Regarding the new variable names, I think they are better than before.

**Review by jonandernovella — APPROVED** [review]
Good job! The rest looks fine to me.

**Review by blankdots — APPROVED** [review]

**Pull request merged**